be/src/olap/rowset/segment_v2/indexed_column_writer.h

```diff
@@ -0,0 +1,104 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "common/status.h"
+#include "gen_cpp/segment_v2.pb.h"
+#include "gutil/macros.h"
+#include "olap/rowset/segment_v2/common.h"
+#include "olap/rowset/segment_v2/page_pointer.h"
+#include "util/slice.h"
+
+namespace doris {
+
+class BlockCompressionCodec;
+class KeyCoder;
+class TypeInfo;
+class WritableFile;
+
+namespace segment_v2 {
+
+class IndexPageBuilder;
+class PageBuilder;
+
+struct IndexedColumnWriterOptions {
+    size_t index_page_size = 64 * 1024;
+    bool write_ordinal_index = false;
+    bool write_value_index = false;
+    EncodingTypePB encoding = DEFAULT_ENCODING;
+    CompressionTypePB compression = NO_COMPRESSION;
+};
+
+// TODO test with empty input (all values are null)
+// TODO support null value
+// TODO support value index with duplicated keys
+class IndexedColumnWriter {
```

Comment:
When I read this class name, I feel that this is the writer for column who has an index.
Can you make it a more clear name?

Reply:
Yeah, the confusion comes from lack of explanation of the `IndexedColumn` concept. I'll add comments for that.